src/Apps/Page/PageApp.tsx

```diff
@@ -25,14 +35,34 @@ const PageApp: FC<PageAppProps> = ({ page }) => {
     }
   }, [page.internalID])
 
-  const { match } = useRouter()
+  useEffect(() => {
+    if (PAGE_SLUGS_WITH_AUTH_REQUIRED.includes(page.internalID) && !user?.id) {
+      showAuthDialog({
+        mode: "SignUp",
+        options: {
+          title: mode => {
+            const action = mode === "SignUp" ? "Sign up" : "Log in"
+            return `${action} to view ${page.name}`
+          },
+          redirectTo: match.location.pathname,
+        },
+        analytics: {
+          contextModule: ContextModule.header,
+        },
+      })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [page, user])
 
   const canonical = TOP_LEVEL_PAGE_SLUG_ALLOWLIST.includes(match.params.id)
     ? `/${match.params.id}`
     : `/page/${match.params.id}`
 
   if (!page.content) return null
 
+  if (PAGE_SLUGS_WITH_AUTH_REQUIRED.includes(page.internalID) && !user?.id)
```

Comment:
#non-blocking. The logic repeats so potentially can be DRY'd with something like `if (auth_required)`.